components/draw/draw.service.ts

```diff
@@ -375,6 +423,41 @@ export class HsDrawService {
         source: this.source,
         type: /** @type {GeometryType} */ this.type,
         style: changeStyle ? changeStyle() : undefined,
+        condition: (e) => {
```

Comment:
Should check if user tries to right click without adding any points on the map, otherwise getting null value.

Reply:
Not sure I understand what do you mean.

Reply:
When I select line or polygon, and click right mouse click on the map, I get error, cannot read length of undefined.

Reply:
https://drive.google.com/file/d/1-g6Kb_CRPaqOcBa6lDvIpczOVP2I9idM/view This is the error I was experiencing.

Reply:
Oh I see, right . Thanks.